### DISCUSSION TITLE
UART log : line feeds not correctly

### DISCUSSION THREAD

**Discussion by yoshinoyatoko** [memtest86plus/memtest86plus#389] [Enhancement]
Hi, everyone:
I used putty to save the uart log output.
I found it seemed displaying all contents in one line, line feed not correctly.

~=~=~=~=~=~=~=~=~=~=~= PuTTY log 2024.03.06 16:20:21 =~=~=~=~=~=~=~=~=~=~=~=
^[[2;35H^[[0m 67% ##########################               ^[[3;35H 70% ############################             ^[[4;35H #6  [Moving inversions, 64 bit pattern]      ^[[5;35Hing: 22GB - 23GB [1GB of 31.9GB]              ^[[6;35Hern: 0xfffffffdffffffff                       ^[[8;1H^[[0mCPU: 4 Cores 4 Threads    SMP: 4T (PAR)   | Time:  0:23:26  Status: Testing  \  ^[[9;1HUsing: All Cores                          | Pass:  0        Errors: 0           ^[[2;17H^[[0mz   68/69*C^[[2;35H^[[0m 67% ##########################               ^[[3;35H 71% ############################             ^[[4;35H #6  [Moving inversions, 64 bit pattern]      ^[[5;35Hing: 22GB - 23GB [1GB of 31.9GB]              ^[[6;35Hern: 0xffffffefffffffff                       ^[[8;1H^[[0mCPU: 4 Cores 4 Threads    SMP: 4T (PAR)

**Comment by tormodvolden** [comment]
The serial console output is a mirror of the normal screen, using terminal emulation codes for viewing in a terminal emulator. This is why you see the ^[[ codes and no line feeds. It is not suitable for saving to a file.

**Comment by yoshinoyatoko** [comment]
Hi, Tormod
Got it, thanks
